### PR TITLE
[WFLY-13350] Add a Galleon layer for JSF support

### PR DESCRIPTION
**Pull request by yersan**
Jira issue: https://issues.jboss.org/browse/WFLY-13350
Proposal: https://github.com/wildfly/wildfly-proposals/pull/310

### CODE REVIEW

**Review by fjuma — APPROVED** [review]

**Comment by yersan** [comment]
Minor review, JSF tests moved under `layers.profile`, the were incorrectly configured in `standalone.microprofile.profile`

**Comment by yersan** [comment]
Hello @fjuma , I asked for a review from your side too early in the process :(, now that we have the AD approved, would you mind taking a look again to this PR as SME and approve if you are fine with it? Thanks!

**Comment by yersan** [comment]
Hi @fjuma , sorry for the noise, issue is still "in Progress". We are still pending to add an additional surefire execution here, so, forget my previews comment.

File: testsuite/integration/basic/pom.xml
Comment (tommaso-borgato) [inline testsuite/integration/basic/pom.xml:1283]: @yersan I had to add the following in order to execute the test **InjectionBeanToPhaseListenerTest** mentioned in the TP:
`<include>org/jboss/as/test/integration/jsf/phaselistener/injectiontarget/bean/InjectionBeanToPhaseListenerTest.java</include>`
is it actually missing or the executuion is linked somewhere else?

**Review by tommaso-borgato — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc
Comment (tommaso-borgato) [inline docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc:130]: @yersan as recent user of these new galleon layers I found it extremely hard to understand how to provision a trimmed server after I cloned the WildFly repository; If this is not on purpose, I suggest adding a small paragraph like the following (not necessarily here and probably with better wording):

_After cloning the WildFly repository locally, you can provision a trimmed server leveraging the galleon cli:
`galleon.sh install wildfly:current/snapshot --layers=jsf,datasources-web-server --dir=my-wildfly-server`_

**Review by tommaso-borgato — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc
Comment (yersan) [reply docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc:130]: Hello @tommaso-borgato I opened this task to take a look at some possible improvements about layers documentation https://issues.redhat.com/browse/WFLY-13744, please add there any comments you think we can add into the documentation, thanks

In the test plan, we discussed how to do it, but I got the impression it is a bit confusing. So, let me add here this note, you don't need to clone the WildFly repo to use the Galleon tools. If you want to use this "in progress" layer, then you need to get this topic branch, but after it is merged you don't need to clone WildFly in you local repository. You could just use `galleon.sh install wildfly:current`.

File: testsuite/integration/basic/pom.xml
Comment (yersan) [reply testsuite/integration/basic/pom.xml:1283]: Good catch!, that's an issue with this topic branch.

Comment (yersan) [reply testsuite/integration/basic/pom.xml:1283]: I've just fixed this adding explicitly this extension to include the missing test.

File: docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc
Comment (tommaso-borgato) [reply docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc:130]: @yersan tnks, I added a comment there ....

**Review by tommaso-borgato — APPROVED** [review]

**Comment by fjuma** [comment]
@yersan Looks good. Just as a sanity check, would it be possible to create a gist that shows the configuration that gets added? (As an example, Darran was experimenting with creating a [gist](https://gist.github.com/darranl/5503feaf0ab547fdde5622bd31de8dad/revisions#diff-c63c6a7a0a1893574ec5ab688330064a) for the Remoting Galleon Layer just to be able to easily see the configuration that gets added.)

**Comment by yersan** [comment]
Thank you @tommaso-borgato for all the work reviewing this and the test plan.

@fjuma I'm not sure how the gist can be more useful to you, since this is a new layer, I compared the changes against the configuration generated by the base-server layer, all layers inherit this one. See the gist [here](https://gist.github.com/yersan/a2f15d6200e4d012dca5e49ed9745157/revisions) and share any feedback, thanks!

**Comment by darranl** [comment]
I think some diffs like these could be useful to visualise the effects of this new layer.

For anyone using JSF we know they would need to start from web-server as they are developing a web application.

This first diff shows what is added by adding the jsf layer to the web server layer:
https://gist.github.com/darranl/291b45883ea8e1887976691fb30f0258/revisions#diff-335c62acd1f75f6b6ef60ef18a2cce85

The JSF layer is defined with a couple of optional dependencies on other layers so this next diff shows JSF added with those optional dependencies excluded:
https://gist.github.com/darranl/291b45883ea8e1887976691fb30f0258/revisions#diff-a80fcd64d33349794c15da799fac0dce

The comparison base-server to jsf will also be useful to keep an eye on how much is starting to come in but that may raise more questions about the web-server layer than the new jsf layer.

**Comment by yersan** [comment]
Thanks @darranl , that comparisons are more useful.

**Comment by fjuma** [comment]
Thanks @darranl and @yersan!

**Review by fjuma — APPROVED** [review]

File: testsuite/layers/pom.xml
Comment (tommaso-borgato) [inline testsuite/layers/pom.xml:2301]: @yersan I think it's not strictly related to your MR but I don't understand the reason why on line [2263](https://github.com/wildfly/wildfly/pull/13330/files#diff-02180aff61cddbfaca58fc7682d79ab4R2263) the `artifactId` isn't parametrized like on line [1604](https://github.com/wildfly/wildfly/pull/13330/files#diff-02180aff61cddbfaca58fc7682d79ab4R1604):

```
<artifactId>wildfly-galleon-pack</artifactId>
```

**Review by tommaso-borgato — CHANGES_REQUESTED** [review]

File: testsuite/layers/pom.xml
Comment (bstansberry) [reply testsuite/layers/pom.xml:2301]: @tommaso-borgato If layers only available in wildfly-galleon-pack are part of what's provisioned then you can't use wildfly-ee-galleon-pack.

**Comment by bstansberry** [comment]
I put the 'hold' label on this in favor of #13532, which is just this commit merged on top of the EJB layers branch (#13524) with a commit added that turns on testing of a combo of the jsf+ejb-lite layers.

**Review by tommaso-borgato — APPROVED** [review]

**Pull request merged**
